src/include/platform/DeviceInstanceInfoProvider.h

```diff
@@ -0,0 +1,124 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <lib/core/CHIPError.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace DeviceLayer {
+
+class DeviceInstanceInfoProvider
+{
+public:
+    DeviceInstanceInfoProvider()          = default;
+    virtual ~DeviceInstanceInfoProvider() = default;
+
+    /**
+     * @brief Obtain the Serial Number from the device's factory data.
+     *
+     * The SerialNumber attribute specifies a human readable serial number
+     *
+     * @param[in, out] buf Buffer to copy string.
+     *                 On CHIP_NO_ERROR return from this function this buffer will be null-terminated.
+     *                 On error CHIP_ERROR_BUFFER_TOO_SMALL there is no guarantee that buffer will be null-terminated.
+     * @param[in] bufSize Size of data, including the null terminator, that can be written to buf.
+     *                    This size should be +1 higher than maximum possible string.
+     *                    This value is valid if the return value is either CHIP_NO_ERROR or CHIP_ERROR_BUFFER_TOO_SMALL.
+     * @returns CHIP_NO_ERROR on success, or another CHIP_ERROR from the underlying implementation
+     *          if access fails.
+     */
+    virtual CHIP_ERROR GetSerialNumber(char * buf, size_t bufSize) = 0;
+
+    /**
+     * @brief Obtain a manufacturing date from the device's factory data.
+     *
+     * The ManufacturingDate attribute specifies the date that the Node was manufactured.
+     *
+     *
+     * @param[out] year Reference to location where manufacturing year will be stored
+     * @param[out] month Reference to location where manufacturing month will be stored
+     * @param[out] day Reference to location where manufacturing day will be stored
```

Comment:
This should really document what the month/day format is (e.g. is it 0-based or 1-based).

Reply:
Ok, I guess according to ISO 8601 these values should be 1-based. I'll add it.